modules/workflow-workflowoperation/src/main/java/org/opencastproject/workflow/handler/workflow/StartWorkflowWorkflowOperationHandler.java

```diff
@@ -104,22 +97,33 @@ public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobConte
       properties.put(key, operation.getConfiguration(key));
     }
 
+    final WorkflowDefinition workflowDefinition;
     try {
       // Get workflow definition
-      final WorkflowDefinition workflowDefinition = workflowService.getWorkflowDefinitionById(
+      workflowDefinition = workflowService.getWorkflowDefinitionById(
               configuredWorkflowDefinition);
-
-      // Start workflow
-      logger.info("Starting '{}' workflow for media package '{}'", configuredWorkflowDefinition,
-              configuredMediaPackageID);
-      workflowService.start(workflowDefinition, mp, properties);
-
     } catch (NotFoundException e) {
       throw new WorkflowOperationException(format("Workflow Definition '%s' not found", configuredWorkflowDefinition));
     } catch (Exception e) {
       throw new WorkflowOperationException(e);
     }
 
+    for (String mpId : asList(configuredMediaPackageIDs)) {
+      // Get media package
+      Opt<MediaPackage> mpOpt = assetManager.getMediaPackage(mpId);
+      if (mpOpt.isNone()) {
+        throw new WorkflowOperationException(format("Media package %s not found", mpId));
```

Comment:
Maybe it would be better to record the error but continue with the other IDs and only then fail? Or do you think a fast fail would be better?

Reply:
we should make that optional on fail_on_error (fast fail if fail-on-error = true, record and continue if fail-on-error=false)